src/probnum/randprocs/markov/continuous/_lti_sde.py

```diff
@@ -0,0 +1,223 @@
+            rv_obtained=rv_obtained,
+            rv=rv,
+            rv_forwarded=rv_forwarded,
+            gain=gain,
+            t=t,
+            _diffusion=_diffusion,
+        )
+
+    @functools.lru_cache(maxsize=None)
+    def discretise(self, dt):
+        """Return a discrete transition model (i.e. mild solution to SDE) using matrix
+        fraction decomposition.
+
+        That is, matrices A(h) and Q(h) and vector s(h) such
+        that the transition is
+
+        .. math:: x | x_\\text{old} \\sim \\mathcal{N}(A(h) x_\\text{old} + s(h), Q(h)) ,
+
+        which is the transition of the mild solution to the LTI SDE.
+        """
+
+        if np.linalg.norm(self.force_vector) > 0:
+            zeros = np.zeros((self.state_dimension, self.state_dimension))
+            eye = np.eye(self.state_dimension)
+            drift_matrix = np.block([[self.drift_matrix, eye], [zeros, zeros]])
+            dispersion_matrix = np.concatenate(
+                (self.dispersion_matrix, np.zeros(self.dispersion_matrix.shape))
+            )
+            ah_stack, qh_stack, _ = _mfd.matrix_fraction_decomposition(
+                drift_matrix, dispersion_matrix, dt
+            )
+            proj = np.eye(self.state_dimension, 2 * self.state_dimension)
+            proj_rev = np.flip(proj, axis=1)
+            ah = proj @ ah_stack @ proj.T
+            sh = proj @ ah_stack @ proj_rev.T @ self.force_vector
+            qh = proj @ qh_stack @ proj.T
+        else:
+            ah, qh, _ = _mfd.matrix_fraction_decomposition(
+                self.drift_matrix, self.dispersion_matrix, dt
+            )
+            sh = np.zeros(len(ah))
+        return discrete.LTIGaussian(
+            ah,
+            sh,
+            qh,
+            forward_implementation=self._forward_implementation_string,
+            backward_implementation=self._backward_implementation_string,
+        )
+
+    def _duplicate(self, **changes):
+        """Create a new object of the same type, replacing fields with values from
+        changes."""
+
+        def replace_key(key):
+            """If the key is part of the desired changes, change appropriately.
+
+            Otherwise, take the current value.
+            """
+            try:
+                return changes[key]
+            except KeyError:
+                return getattr(self, key)
+
+        drift_matrix = replace_key("drift_matrix")
+        dispersion_matrix = replace_key("dispersion_matrix")
+        force_vector = replace_key("force_vector")
+        forward_implementation = replace_key("forward_implementation")
+        backward_implementation = replace_key("backward_implementation")
+
+        return LTISDE(
+            drift_matrix=drift_matrix,
+            dispersion_matrix=dispersion_matrix,
+            force_vector=force_vector,
+            forward_implementation=forward_implementation,
+            backward_implementation=backward_implementation,
```

Comment:
I wish we could avoid writing the same code over and over again, for every subclass of `Transition`. But from the top of my head, I cannot think of a better solution.

Reply:
Maybe something along the lines of
```python
import copy

# ...

def _duplicate(self, **changes):
  lti_kwargs = copy.deepcopy(self.__dict__)
  lti_kwargs.update(changes)
  return LTISDE(**lti_kwargs)
```

If in `changes` there is an invalid key, then I think there will be this `received unexpected argument` error.

Reply:
I agree with you. Maybe there is a better solution. Unfortunately, your suggestion above doesnt work (because the elements in __dict__ are not directly related to the signature of __init__). Like you, I thought about copy.deepcopy, but wasnt able to make it work the way I wanted.

I will try to do some brainstorming for a more elegant solution. Off the top of my head, perhaps some dispatchy thing? Or maybe even make the whole thing a dataclass? Dunno...